cou/apps/base.py

```diff
@@ -624,6 +624,58 @@ def _get_disable_action_managed_plan(self, parallel: bool = False) -> UpgradeSte
             )
         return UpgradeStep()
 
+    def _get_enable_action_managed_plan(self, parallel: bool = False) -> UpgradeStep:
+        """Get plan to enable action-managed-upgrade.
+
+        This is used to upgrade as "paused-single-unit" strategy.
+
+        :param parallel: Parallel running, defaults to False
+        :type parallel: bool, optional
+        :return: Plan to enable action-managed-upgrade
+        :rtype: UpgradeStep
+        """
+        if self.config.get("action-managed-upgrade", {}).get("value", False):
+            return UpgradeStep()
+        return UpgradeStep(
+            description=(
+                f"Change charm config of '{self.name}' 'action-managed-upgrade' to True."
+            ),
+            parallel=parallel,
+            coro=self.model.set_application_config(self.name, {"action-managed-upgrade": True}),
+        )
+
+    def _get_pause_unit(self, unit: ApplicationUnit, parallel: bool = False) -> UpgradeStep:
+        """Get the plan to pause a unit to upgrade.
+
+        :param unit: Unit to be paused.
+        :type unit: ApplicationUnit
+        :param parallel: Parallel running, defaults to False
+        :type parallel: bool, optional
+        :return: Plan to pause a unit.
+        :rtype: UpgradeStep
+        """
+        return UpgradeStep(
```

Comment:
We should check if this action exits or not? Or maybe define this only in application with such action.
For a situation like this, it would be nice to have mixing classes.

Reply:
We will handle it later or maybe not care.